kunta-api-server/src/test/java/fi/otavanopisto/kuntaapi/test/AbstractTest.java

```diff
@@ -0,0 +1,157 @@
+package fi.otavanopisto.kuntaapi.test;
+
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+public abstract class AbstractTest {
+  
+  protected int getHttpPort() {
+    return Integer.parseInt(System.getProperty("it.port.http"));
+  }
+
+  protected String getHost() {
+    return System.getProperty("it.host");
+  }
+  
+  protected int getWireMockPort() {
+    return getHttpPort() + 1;
+  }
+  
+  protected String getWireMockBasePath() {
+    return String.format("http://%s:%d", getHost(), getWireMockPort());
+  }
+
+  protected String getApiBasePath() {
+    return String.format("http://%s:%d/v1", getHost(), getHttpPort());
+  }
+  
+  protected long insertOrganizationSetting(String organizationId, String key, String value) {
+    return executeInsert("insert into OrganizationSetting (settingKey, organizationKuntaApiId, value) values (?, ?, ?)", key, organizationId, value);
+  }
+
+  protected long insertSystemSetting(String key, String value) {
+    return executeInsert("insert into SystemSetting (settingKey, value) values (?, ?)", key, value);
+  }
+  
+  protected void deleteOrganizationSetting(String organizationId, String key) {
+    executeDelete("delete from OrganizationSetting where settingKey = ? and organizationKuntaApiId = ?", key, organizationId);
+  }
+
+  protected void deleteSystemSetting(String key) {
+    executeDelete("delete from SystemSetting where settingKey = ?", key);
+  }
+
+  protected long createIdentifier(String kuntaApiId, String source, String sourceId, String type) {
+    return executeInsert("insert into Identifier (kuntaApiId, source, sourceId, type) values (?, ?, ?, ?)", 
+      kuntaApiId, source, sourceId, type);
+  }
+
+  protected void deleteIndentifier(String key) {
+    executeDelete("delete from Identifier where kuntaApiId = ?", key);
+  }
+  
+  protected long executeInsert(String sql, Object... params) {
+    try {
+      Connection connection = getConnection();
+      try {
+        connection.setAutoCommit(true);
+        PreparedStatement statement = connection.prepareStatement(sql);
+        try {
+          for (int i = 0, l = params.length; i < l; i++) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)